compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -2477,18 +2477,35 @@ TR::Instruction *
 genLoadAddressConstant(TR::CodeGenerator * cg, TR::Node * node, uintptr_t value, TR::Register * targetRegister,
    TR::Instruction * cursor, TR::RegisterDependencyConditions * cond, TR::Register * base)
    {
-   TR_ExternalRelocationTargetKind reloKind = TR_NoRelocation;
-   if (cg->profiledPointersRequireRelocation() &&
-            node->getOpCodeValue() == TR::aconst &&
-            (node->isMethodPointerConstant() || node->isClassPointerConstant()))
+   bool isPicSite = false;
+   bool assumeUnload = false;
+   bool isMethodPointer = false;
+   if (node->getOpCodeValue() == TR::aconst)
```

Comment:
Hi, coming back to this PR. I think we should carefully evaluate the way we are generating instructions here. 
Looking at the changes, one thing pops in my mind is that what is the difference between `isUnloadAssumptionRequired` and `node->isClassUnloadingConst()`? Looking at the implementation, it does seem like `isUnloadAssumptionRequired` would check the class of the method and passed in class pointer to see if it is a constant that needed to be updated upon class unloading. While `node->isClassUnloadingConst()` seems to check if constant value of  Literal Pool/aconst Node needs to be patched upon class unloading. 

I wonder, if we need to use only one ? 

Also I think this can be refactored further to tackle the cases where we would go to path with relocation. 

Just a question as I looked into this change again after a month.

Reply:
I dug into the implementations of `isUnloadAssumptionRequired()` and `node->isClassUnloadingConst()`. `isUnloadAssumptionRequired()` looks like a broader check, and given that the other platforms don't use `isClassUnloadingConst` in while generating address constants it should be safe to remove the `isClassUnloadingConst` check. I ran sanity tests with the check removed and didn't see any failures.